storage/dataflux/worksteal.go

```diff
@@ -45,16 +56,171 @@ type worker struct {
 	idleChannel   chan int
 	result        *listerResult
 	generation    int64
+	lister        *Lister
 }
 
 // workstealListing is the main entry point of the worksteal algorithm.
 // It performs worksteal to achieve highly dynamic object listing.
-func (c *Lister) workstealListing(ctx context.Context) []*storage.ObjectAttrs {
+// workstealListing creates multiple (parallelism) workers that simultaneosly lists
+// objects from the buckets.
+func (c *Lister) workstealListing(ctx context.Context) ([]*storage.ObjectAttrs, error) {
+	var workerErr []error
+	// Idle channel is used to track number of idle workers.
+	idleChannel := make(chan int, c.parallelism)
+	// Result channel is used to track the results from each worker.
+	result := &listerResult{
```

Comment:
This is not a channel, right?

Reply:
Its not a channel. Thanks for catching this. 
Updated the comment.